File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/features/auth/login/LoginRequest.java
package br.com.deltaglobalbank.identity.features.auth.login;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

public record LoginRequest(
    @NotBlank(message = "email é obrigatório")
    @Email(message = "email inválido")
    String email,

    @NotBlank(message = "password é obrigatório")
    String password
) {
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/features/auth/login/LoginRequest.kt
package br.com.deltaglobalbank.identity.features.auth.login

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

data class LoginRequest(
    @field:NotBlank(message = "email é obrigatório")
    @field:Email(message = "email inválido")
    val email: String,

    @field:NotBlank(message = "password é obrigatório")
    val password: String
)